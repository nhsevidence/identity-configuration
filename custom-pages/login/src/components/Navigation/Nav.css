@import '../variables.css';


.activeRoute {
  font-weight: bold;
  text-decoration: none;
}

.navigation{
    text-align: center;
    width: var(--col-width);
}

.navigation a:hover{
    text-decoration: none;
}
