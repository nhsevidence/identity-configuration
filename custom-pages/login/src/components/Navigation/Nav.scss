@import '../variables.scss';


.activeRoute {
  font-weight: bold;
  text-decoration: none;
}

.navigation{
    // text-align: center;
    width: var(--col-width);
}

.navigationLink{
  font-weight: bold;
  font-size: 1rem;
  margin: 1rem;
}

.navigation span{
  padding-right: 5px;
}

.navigation a:hover{
    text-decoration: none;
}
