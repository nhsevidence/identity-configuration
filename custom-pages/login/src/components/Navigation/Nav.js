import React from 'react'
// import PropTypes from 'prop-types'
import { IndexLink, Link } from 'react-router'
import './Nav.scss'

export const Navigation = () => {
  const register = window.location.href.indexOf('register') !== -1
  return (
    <div className="navigation">
      {register ? (
        <div className="navigationLink">
          <h3> Register </h3>
          <span>Already have an NICE Account?</span>
          <IndexLink to="/" activeClassName="activeRoute">
            Sign in
          </IndexLink>
        </div>
      ) : (
        <div className="navigationLink">
          <h3> Log in </h3>
          <span>{'Don\'t have an NICE Account?'}</span>
          <Link data-qa-sel="Signin-link-login" to="/register" activeClassName="activeRoute">
            Sign up
          </Link>
        </div>
      )}
    </div>
  )
}

// Navigation.propTypes = {
//   children: PropTypes.element.isRequired
// }

export default Navigation
