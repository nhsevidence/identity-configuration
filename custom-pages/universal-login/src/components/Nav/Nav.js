import React from 'react'
import { NavLink, Link } from 'react-router-dom'
import './Nav.scss'

const Nav = () => {
  const register = window.location.pathname.indexOf('register') !== -1;
  return (
    <div className="navigation">
      {register ? (
        <div className="navigationLink">
          <h3> Create account </h3>
          <p className="lead"><NavLink
            data-qa-sel="Signin-link-login"
            to="/"
            activeclassname="activeRoute"
          >
            Sign in
          </NavLink> if you already have a NICE account.</p>
        </div>
      ) : (
        <div className="navigationLink">
          <h3> Log in </h3>
          <Link
            data-qa-sel="Signup-link-login"
            to="/register"
            activeclassname="activeRoute"
          >
            Create a NICE account
          </Link>
        </div>
      )}
    </div>
  )
}

export default Nav
