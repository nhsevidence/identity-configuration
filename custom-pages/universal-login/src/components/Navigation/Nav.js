import React from 'react'
// import PropTypes from 'prop-types'
import { NavLink, Link } from 'react-router-dom'
import './Nav.scss'

export const Navigation = () => {
  const register = window.location.href.indexOf('register') !== -1;

  return (
    <div className="navigation">
      {register ? (
        <div className="navigationLink">
          <h3> Create account </h3>
          <span>Sign in if you already have a NICE account.</span>
          <NavLink
            data-qa-sel="Signin-link-login"
            to="/"
            activeclassname="activeRoute"
          >
            Sign in
          </NavLink>
        </div>
      ) : (
        <div className="navigationLink">
          <h3> Log in </h3>
          <span>Need a NICE Account?</span>
          <Link
            data-qa-sel="Signup-link-login"
            to="/register"
            activeclassname="activeRoute"
          >
            Create an account
          </Link>
        </div>
      )}
    </div>
  )
}

// Navigation.propTypes = {
//   children: PropTypes.element.isRequired
// }

export default Navigation
