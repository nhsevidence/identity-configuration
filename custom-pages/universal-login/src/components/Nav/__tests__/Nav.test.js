import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Nav from '../Nav'

Enzyme.configure({ adapter: new Adapter() });

describe('Nav components', () => {
  it('should render Nav correctly', () => {
    expect(shallow(<Nav />)).toMatchSnapshot()
  })
});
