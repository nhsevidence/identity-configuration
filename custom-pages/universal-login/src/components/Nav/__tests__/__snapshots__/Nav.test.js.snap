// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav components should render Nav correctly 1`] = `ShallowWrapper {}`;
